File: hackerrank/easy/0048-minimum-distances/python/solution.py
## solution.py

from typing import List, Dict

def minimumDistances(a: List[int]) -> int:
    last: Dict[int, int] = {}
    min_d = None

    for i, v in enumerate(a):
        if v in last:
            d = i - last[v]
            min_d = d if min_d is None or d < min_d else min_d
            if min_d == 1:
                return 1  # early exit
        last[v] = i

    return -1 if min_d is None else min_d

# --- demo run ---
if __name__ == "__main__":
    demo1 = [7, 1, 3, 4, 1, 7]
    print(minimumDistances(demo1))  # Expected: 3

    demo2 = [1, 2, 3]
    print(minimumDistances(demo2))  # Expected: -1

    demo3 = [5, 5, 5]
    print(minimumDistances(demo3))  # Expected: 1

File: hackerrank/easy/0048-minimum-distances/js/solution.mjs
// solution.mjs

export function minimumDistances(a) {
  const last = new Map();
  let min = Infinity;

  for (let i = 0; i < a.length; i++) {
    const v = a[i];
    if (last.has(v)) {
      const d = i - last.get(v);
      if (d < min) min = d;
      if (min === 1) return 1; // early exit
    }
    last.set(v, i);
  }

  return min === Infinity ? -1 : min;
}

// --- demo run ---
const demo1 = [7, 1, 3, 4, 1, 7];
console.log(minimumDistances(demo1)); // Expected: 3

const demo2 = [1, 2, 3];
console.log(minimumDistances(demo2)); // Expected: -1

const demo3 = [5, 5, 5];
console.log(minimumDistances(demo3)); // Expected: 1